can/tests/test_can_parser.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+import glob
+import os
+import time
+import unittest
+
+from opendbc import DBC_PATH
+from opendbc.can.packer import CANPacker
+from opendbc.can.parser import CANParser
+from opendbc.can.tests.test_packer_parser import can_list_to_can_capnp
+from opendbc.generator.generator import generated_suffix
+
+
+class TestCANParser(unittest.TestCase):
+  @classmethod
+  def setUpClass(cls):
+    cls.dbcs = []
+    for dbc in glob.glob(f"{DBC_PATH}/*{generated_suffix}"):
+      cls.dbcs.append(os.path.basename(dbc).split('.')[0])
+
+  def test_detecting_dbcs(self):
+    self.assertGreaterEqual(len(self.dbcs), 20)
+
+  def test_dbc_parsing_speed(self):
+    start_time = time.time()
+    for dbc in self.dbcs:
+      CANParser(dbc, [], [], 0)
+
+    elapsed = time.time() - start_time
+    self.assertLess(elapsed, 0.15, "Took too long to parse {} DBCs".format(len(self.dbcs)))
+
+  def test_can_parsing_speed(self):
```

Comment:
this doesn't really go in this PR. I'd suggest breaking out this test class into another PR merged before this one and only adding relevant test cases in this PR

Reply:
Wanted to test changing from arrays to vectors, but agreed. No change in time here